src/tests/system/tests/test_access_control_ldap_filter.py

```diff
@@ -0,0 +1,251 @@
+"""
+SSSD Authentication Test Cases
+
+:requirement: access control access_filter
+
+LDAP Filters are not intenteded to work with IPA provider
+(there are other mechanisms in IPA to handle it way better than ldap filter),
+thus we are skipping IPA provider
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.topology import KnownTopologyGroup
+from sssd_test_framework.roles.ad import AD
+from sssd_test_framework.roles.samba import Samba
+from sssd_test_framework.roles.ldap import LDAP
+from sssd_test_framework.roles.ipa import IPA
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+@pytest.mark.require(lambda provider: not isinstance(provider, IPA), "It is not testable with IPA")
+def test_access_filter__single_ldap_attribute_permits_user_login(client: Client, provider: GenericProvider):
+    """
+    :title: LDAP attribute filter permits specific user login
+    :description: Verifies that SSSD allows login only for users matching an specific LDAP attribute.
+    :setup:
+        1. Create users ‘user1’ and ‘user2’
+        2. Configure SSSD with ‘access_provider = ldap|ad’ and ‘*_access_filter = uid|samAccountName = user1’
+        3. Start SSSD
+    :steps:
+        1. Attempt login with `user1`.
+        2. Attempt login with `user2`.
+    :expectedresults:
+        1. `user1` is successfully authenticated.
+        2. `user2` is denied access.
+    :customerscenario: False
+    """
+
+    provider.user("user1").add()
+    provider.user("user2").add()
+
+    if isinstance(provider, (AD, Samba)):
+        client.sssd.domain["access_provider"] = "ad"
+        client.sssd.domain["ad_access_filter"] = "(samAccountName=user1)"
+    else:
+        client.sssd.domain["access_provider"] = "ldap"
+        client.sssd.domain["ldap_access_filter"] = "(uid=user1)"
+
+    client.sssd.start()
+
+    assert client.auth.ssh.password("user1", "Secret123"), "User1 should be able to login!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "User2 should NOT be able to login!"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+@pytest.mark.require(lambda provider: not isinstance(provider, IPA), "It is not testable with IPA")
+@pytest.mark.require(
+    lambda provider: not isinstance(provider, LDAP), "Skipping LDAP until we decide what to do with this test"
+)
+def test_access_filter__group_attributes_permits_user_login(client: Client, provider: GenericProvider):
```

Comment:
So this test should be cloned to _test_ldap.py_ and only cover _rfc2307bis_, named **test_ldap__rfc2307bis_group_attribute_permits_user_login**. I know this name may seem strange, but we can filter out these tests quickly, and the name of the test is the same.

Reply:
I created https://github.com/SSSD/sssd-test-framework/issues/164 to handle the memberOf issue we discovered yesterday.

What should I? LDAP filter will not handle memberOf relationship without rfc2307bis here. Should I skip LDAP?

Reply:
I made a decision for now and I decided to run only AD/Samba and made the adjustments. If you are ok with that, we can go ahead and handle rfc2307bis in ldap.py (I mentioned in the test as well).